src/plugins/data/common/index_patterns/expressions/load_index_pattern.ts

```diff
@@ -57,4 +58,24 @@ export const getIndexPatternLoadMeta = (): Omit<
       }),
     },
   },
+  extract(state) {
+    const refName = 'indexPatternLoad.id';
+    const references: SavedObjectReference[] = [
+      {
+        name: refName,
+        type: 'search',
+        id: state.id[0] as string,
+      },
+    ];
+    state.id[0] = refName;
```

Comment:
Is that okay that the state object is mutable? Wouldn't it cause any side effects?

Reply:
i'll  update it to not mutate